driver-core/src/main/java/com/datastax/driver/core/SessionManager.java

```diff
@@ -376,8 +376,14 @@ public void onFailure(Throwable t) {
                             cluster.manager.triggerOnDown(host, false);
                         } else {
                             logger.warn("Error creating pool to " + host, t);
+                            cluster.manager.triggerOnDown(host, true);
```

Comment:
I don't think this is necessary. It should be the conviction policy's job to mark the node down if there are no more connections to it.
If there are still active connections (for example from another session, or the control connection) then the host should stay up.